src/cleanup_rules/swift/rules.toml

```diff
@@ -649,14 +654,201 @@ queries = [
 
   # skip the rule if it is accessed in the self.x manner
   """(
+        (navigation_expression
+            target: (self_expression)
+            suffix: (navigation_suffix
+                suffix: (simple_identifier) @variable
+            )
+        )
+        (#eq? @variable "@identifier")
+    )""",
+]
+
+[[rules]]
+name = "delete_field_declaration"
+query = """(
+            (property_declaration
+                name: (pattern
+                    bound_identifier: (simple_identifier) @avariable
+                )
+            ) @property_declaration
+    (#eq? @avariable "@hvariable")
+)"""
+replace_node = "property_declaration"
+replace = ""
+holes = ["hvariable"]
+is_seed_rule = false
+
+[[rules.constraints]]
+matcher = "(class_declaration) @ccd"
+# rule to replace field variable used as self.x subject to the mentioned constraints
```

Comment:
this comment could be moved above l.667?

Reply:
Can't be, different rules. But added a comment there too.